templates/raspbian_cattlepi/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers

```diff
@@ -355,3 +355,60 @@ cattlepi_build_root_filesystem()
     echo $CATTLEPI_BASE > "${rootmnt}/cattlepi/base"
     echo "boot/${CATTLEPI_ID}/config" > "${rootmnt}/cattlepi/base_relative_config"
 }
+
+cattlepi_bring_up_wlan0() {
+    interface="wlan0"
+    wait_limit="30"
+    online=false
+
+    echo "Bringing up wlan0"
+
+    if [ ! -x "/sbin/wpa_supplicant" ]; then
+        echo "/sbin/wpa_supplicant doesn't exist, exiting wlan config"
+        return 1
+    fi
+
+    echo "Starting wireless connection"
+    /sbin/wpa_supplicant  -i"${interface}" -c/etc/wpa_supplicant/wpa_supplicant.conf -P/run/initram-wpa_supplicant.pid -B -f /tmp/wpa_supplicant.log
```

Comment:
The conf is still to be generated, right?

Reply:
The conf is to be pulled down from the API.  After the first boot, if a conf exists we will persist it in order to use it on future boots.

Reply:
Right now if the conf changes and autouodate is true it will reboot to pickup the new changes. Not in the scope of this work, but maybe we can update the configuration (on sd card) before rebooting to enable scenarios where we don’t have connectivity on boot. Just a thought - and not a really well baked one

Reply:
Yeah that makes sense.  I'll see if I can integrate that into the second stage of this work.